free-programming-books.md

```diff
@@ -1418,15 +1421,15 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Java Language and Virtual Machine Specifications](https://docs.oracle.com/javase/specs/) - James Gosling, et al
 * [Java Notes for Professionals](http://books.goalkicker.com/JavaBook/) - Compiled from StackOverflow documentation (3.x)
 * [Java Programming for Kids](https://yfain.github.io/Java4Kids/) - Yakov Fain
-* [Java: The Legend](https://www.oreilly.com/programming/free/java-the-legend.csp) - Ben Evans (email address *requested*, not required)
+* [Java: The Legend](http://pepa.holla.cz/wp-content/uploads/2016/10/java-the-legend.pdf) - Ben Evans (PDF)
```

Comment:
This doesn't appear to be an authorized download.

Reply:
I think we need to discuss more for it in Issue.